liquibase-standard/src/main/java/liquibase/snapshot/JdbcDatabaseSnapshot.java

```diff
@@ -276,9 +276,46 @@ public List<CachedRow> fastFetch() throws SQLException, DatabaseException {
                         }
                     }
 
+                    enrichPostgresqlResult(catalogAndSchema, returnList);
+
                     return returnList;
                 }
 
+                private void enrichPostgresqlResult(CatalogAndSchema catalogAndSchema, List<CachedRow> returnList) throws DatabaseException, SQLException {
+                    if (database instanceof PostgresDatabase) {
+                        StringBuilder sql = new StringBuilder("SELECT ns.nspname as TABLE_SCHEMA, tab.relname as TABLE_NAME, " +
+                                "cls.relname as INDEX_NAME, am.amname as INDEX_TYPE " +
+                                " FROM pg_index idx " +
+                                "         JOIN pg_class cls ON cls.oid=idx.indexrelid " +
+                                "         JOIN pg_class tab ON tab.oid=idx.indrelid " +
+                                "         JOIN pg_am am ON am.oid=cls.relam " +
+                                "         JOIN pg_namespace ns ON ns.oid=cls.relnamespace " +
+                                "WHERE ns.nspname = '").append(database.correctObjectName(catalogAndSchema.getSchemaName(), Schema.class)).append("'");
+
+                        if (!isBulkFetchMode && tableName != null) {
+                            sql.append(" AND tab.relname = '").append(database.escapeStringForDatabase(tableName)).append("'");
+                        }
+
+                        if (!isBulkFetchMode && indexName != null) {
+                            sql.append(" AND cls.relname='").append(database.escapeStringForDatabase(indexName)).append("'");
+                        }
+
+                        List<CachedRow> usingTypeInformation = executeAndExtract(sql.toString(), database);
+
+                        for (CachedRow row : usingTypeInformation) {
+                            // Postgres returns the index type for the using field, so we need to set it to "INDEX_TYPE" in the matching entries from the returnList
+                            for (CachedRow returnRow : returnList) {
+                                if (returnRow.getString("INDEX_NAME").equalsIgnoreCase(row.getString("INDEX_NAME"))
+                                    && returnRow.getString("TABLE_NAME").equalsIgnoreCase(row.getString("TABLE_NAME"))
+                                    && returnRow.getString("TABLE_SCHEM").equalsIgnoreCase(row.getString("TABLE_SCHEMA"))) {
```

Comment:
Can't we do this filtering in the SQL statement, instead of in memory in Java? This could have performance implications for large result sets.

Reply:
The filtering is done in the SQL when we are not in `isBulkFetchMode` .  In this mode the only other option would be to run this query for every single `index`... we would save memory but perform many select requests instead.